### PR TITLE
Sdk test projects

### PR DESCRIPTION
**Pull request by chsienki**
New PR replacing #18810 rebased on top of #18695:
___
This change updates the associated targets and props files in the test directory to allow the test csproj's to be built as SDK style projects alongside traditional style projects.

It does not yet upgrade any of the projects, I'll submit a seperate batch of PRs for that, so that these changes aren't lost in a ton of mechanical refactoring.

While these changes allow the csproj's to be upgraded, there is still work to be done to allow depproj and ilproj's to be converted over to the new style. By allowing a mix of projects styles this change allows us to upgrade piece-meal rather than in a single big bang, hopefully allowing us to easier figure out bugs if they arise. 

- Remove CodeTaskFactory:
  - Allows the projects to be built using the core version of msbuild/dotnet build
  - Converted to using msbuild property expansion instead
- Add directory.build.{props,targets}:
  - Currently we just import the dirs.props and targets, but means SDK style projects don't need to explicitly include these files
   - We probably want to move all projects over to using these in the future, but this keeps the changes smaller for now
- Specific code for SDK projects:
  - There are a several changes required to build an SDK project. This change guards them behind conditionals so that only the new style projects see them. When we get to the point that there are only new projects, we can remove the guards (probably at the same time as ditching the dir.props)
- Reordered build targets:
  - Because SDK projects implicitly import the build targets, we can no longer re-define the build targets unconditionally knowing they will likely be overwritten. 
  - Instead we move the overwritten targets to seperate files, and include these conditionally based on properties. In this way there is always a build defined for SDK projects, which can then be overwritten to do nothing as needed.

### CODE REVIEW

File: tests/src/runonly.targets [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <Target Name="CreateManifestResourceNames" />
+  <Target Name="Compile" />
+  <Target Name="CoreCompile" />
+
+  <!-- if this is a RunOnly project that has been scheduled to build, make the output directory and build any dependent projects -->
+  <Target Name="Build" Condition="('$(CLRTestKind)'=='RunOnly') And '$(_WillCLRTestProjectBuild)'">
+    <MSBuild Projects="@(ProjectReference)" />
+    <MakeDir ContinueOnError="false" Directories="$(OutputPath)" />
```
Comment (AaronRobinsonMSFT) [inline tests/src/runonly.targets]: I don't remember the default behavior of this, but does the `MakeDir` task consider a directory that already exists an error?

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Innerloop Build
@dotnet-bot test OSX10.12 x64 Checked CoreFX Tests 
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test all please

File: tests/src/runonly.targets (hunk 1 above)
Comment (chsienki) [reply tests/src/runonly.targets]: It continues if it already exists: the [docs](https://msdn.microsoft.com/en-us/library/s2448zz7.aspx) aren't clear, but looking at the [code](https://github.com/Microsoft/msbuild/blob/7d8c8da147252ea57f35d2c6ceee8e7c5ba458f5/src/Tasks/MakeDir.cs#L58) you can see it just does nothing if it's already there. 

 This is actually moved code from before, so the behavior should stay the same as previously.

**Comment by chsienki** [comment]
/cc @weshaggard @RussKeldorph 

Everything look ok to you?

**Pull request merged**